validator.py

```diff
@@ -563,6 +565,36 @@ def simple_value_matching(self):
                         for key in self.dep_json['simpleAllowedValues'][field][val]:
                             self.search_for_key(True, found, [key], "is '" + val + "'", self.dep_json['simpleAllowedValues'][field][val][key])
 
+
+    def require_unstructured(self):
+        '''
+        Within every scenes[].state, at least one unstructured field must be provided.
+        '''
+        data = copy.deepcopy(self.loaded_yaml)
+        i = 0
+        for scene in data['scenes']:
+            if 'state' in scene:
+                state = scene['state']
+                # look for an unstructured field
+                found = self.find_unstructured(state)
+                if not found:
+                    # unstructured not found - error
+                    self.logger.log(LogLevel.WARN, "At least one 'unstructured' key must be provided within each scenes[].state but is missing at scene[" + str(i) + "]")
+                    self.missing_keys += 1
+            i += 1
+
+    def find_unstructured(self, obj):
+        '''
+        Looks through obj for an unstructured field
+        '''
+        found = False
+        for k in obj:
```

Comment:
I think you need to check for None before iterating.
Dumb edge case, but if I remove everything from the `state` object in a scene (i.e., I include an empty state object), I get a crash:  `TypeError: 'NoneType' object is not iterable`

Reply:
I don't like that. I feel like Python should not consider "None" an instance of a dictionary. At least it's an easy fix, thanks for finding it!